src/adapters/pnp/Watcher.c

```diff
@@ -259,6 +267,30 @@ void InitializeWatcher(const char* jsonConfiguration, void* log)
     RestrictFileAccessToCurrentAccountOnly(DC_FILE);
     RestrictFileAccessToCurrentAccountOnly(RC_FILE);
     RestrictFileAccessToCurrentAccountOnly(GIT_DC_FILE);
+
+    if (NULL != g_mpiHandle)
+    {
+        CallMpiClose(g_mpiHandle, GetLog());
+        g_mpiHandle = NULL;
+    }
+
+    if (true == EnableAndStartDaemon(g_mpiServer, GetLog()))
+    {
+        sleep(1);
```

Comment:
Is this the most preferred way to determine if the service is ready?

Reply:
It's not about service being ready. That's waiting a bit between starting the service (if not started already) and calling MpiOpen. Standard.